projects/lights_control/src/lights_config.c

```diff
@@ -0,0 +1,121 @@
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_FRONT_LIGHT_LEFT_TURN) |             //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_FRONT_LIGHT_SIDE_RIGHT_INDICATOR) |  //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_FRONT_LIGHT_RIGHT_TURN) },           //
+};
+
+static uint16_t s_rear_event_mappings[][2] = {
+  { LIGHTS_EVENT_STROBE_BLINK,                                               //
+    LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_STROBE) },                //
+  { LIGHTS_EVENT_BRAKES,                                                     //
+    LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_BRAKE) |            //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_OUTER_BRAKE) |  //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_BRAKE) |         //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_OUTER_BRAKE) |   //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_CENTRE_BRAKE) },      //
+  { LIGHTS_EVENT_SIGNAL_LEFT_BLINK,                                          //
+    LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_OUTER_TURN) |        //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_TURN) },         //
+  { LIGHTS_EVENT_SIGNAL_RIGHT_BLINK,                                         //
+    LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_OUTER_TURN) |       //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_TURN) },        //
+  { LIGHTS_EVENT_SIGNAL_HAZARD_BLINK,                                        //
+    LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_OUTER_TURN) |        //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_LEFT_TURN) |          //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_OUTER_TURN) |   //
+        LIGHTS_CONFIG_TO_MASK(LIGHTS_CONFIG_REAR_LIGHT_RIGHT_TURN) },        //
+};
+
+static LightsConfig s_config_front = {
+  .board_type = LIGHTS_CONFIG_BOARD_TYPE_FRONT,                  //
+  .addresses = s_addresses_front,                                //
+  .num_addresses = SIZEOF_ARRAY(s_addresses_front),              //
+  .event_mappings = s_front_event_mappings,                      //
+  .num_supported_events = SIZEOF_ARRAY(s_front_event_mappings),  //
+};
+
+static LightsConfig s_config_rear = {
+  .board_type = LIGHTS_CONFIG_BOARD_TYPE_REAR,                  //
+  .addresses = s_addresses_rear,                                //
+  .num_addresses = SIZEOF_ARRAY(s_addresses_rear),              //
+  .event_mappings = s_rear_event_mappings,                      //
+  .num_supported_events = SIZEOF_ARRAY(s_rear_event_mappings),  //
+};
+
+static LightsConfig *s_config;
+
+StatusCode lights_config_init(void) {
+  GPIOSettings settings = {
+    .direction = GPIO_DIR_IN,        //
+    .resistor = GPIO_RES_NONE,       //
+    .alt_function = GPIO_ALTFN_NONE  //
+  };
+
+  // initializing board type pin
+  status_ok_or_return(gpio_init_pin(&s_board_type_address, &settings));
```

Comment:
You know if you made this a function instead that this module takes as an input, it would be really easy to change or mock.

Reply:
like have the `lights_config_init` function take in another function pointer, and that function will implement the procedure of determining which board we are on?